compiler/src/dmd/backend/cod2.d

```diff
@@ -2408,13 +2408,10 @@ void cdcond(ref CodeBuilder cdb,elem *e,regm_t *pretregs)
             codelem(cdb,e1,&retregs,false);
             const reg = findreg(retregs);
 
-            // https://issues.dlang.org/show_bug.cgi?id=23743
-            // Optimization has bugs where it negates the wrong register,
-            // disable for now
-            if (false && v1 == 0 && v2 == ~cast(targ_size_t)0)
+            if (v1 == 0 && v2 == ~cast(targ_size_t)0)
             {
                 cdb.gen2(0xF6 + (opcode & 1),grex | modregrmx(3,2,reg));  // NOT reg
-                if (I64 && sz2 == REGSIZE)
+                if (I64 && sz2 == 1 && reg >= 4)
                     code_orrex(cdb.last(), REX_W);
```

Comment:
No. Leave 2417 and 2418 intact. Insert the following after 2418:
```
        if (I64 && sz2 == 1 && reg >= 4)
            code_orrex(cdb.last(), REX);
```

Reply:
Note that this pattern is used in the surrounding code. Whenever you see an odd 'H' register in the codegen output, this is the fix.

Reply:
Thanks. It's a bit concerning the test case isn't red because of this.